verticapy/tests/vDataFrame/test_vDF_plot_plotly.py

```diff
@@ -1287,3 +1302,133 @@ def test_additional_options_custom_height(self, load_plotly, titanic_vd):
         result = titanic_vd.outliers_plot(columns=["age", "fare"], height=custom_height)
         # Assert
         assert result.layout["height"] == custom_height, "Custom height not working"
+
+
+class TestVDFACFPlot:
+    def test_properties_output_type_for(self, load_plotly, amazon_vd):
+        # Arrange
+        # Act
+        result = amazon_vd.acf(
+            ts="date",
+            column="number",
+            p=48,
+            by=["state"],
+            unit="month",
+            method="spearman",
+        )
+        # Assert - checking if correct object created
+        assert type(result) == plotly.graph_objs._figure.Figure, "wrong object crated"
+
+    def test_properties_xaxis_label(self, load_plotly, amazon_vd):
+        # Arrange
+        test_title = "Lag"
+        # Act
+        result = amazon_vd.acf(
```

Comment:
you're computing act in each of the sub function making it very expensive. Why not computing it one time and sharing it between the functions?

Reply:
optimized as much as possible for ACF